packages/dd-trace/src/appsec/iast/taint-tracking/taint-tracking-impl.js

```diff
@@ -110,7 +114,14 @@ function csiMethodsOverrides (getContext) {
       getContext,
       String.prototype.trim,
       String.prototype.trimStart
-    )
+    ),
+
+    random: function (res, fn, target) {
+      if (mathRandomCallCh.hasSubscribers) {
+        mathRandomCallCh.publish({ target })
```

Comment:
Why send target and not fn? We want to detect vulnerability when `fn === Math.random`, `target` doesn't matter.

Reply:
Yep, way more clean your way. I've applied the modification to check the fn instead of the target obj.